docs/advanced/UsageWithReactRouter.md

```diff
@@ -1,4 +1,158 @@
 # Usage with React Router
 
-Sorry, but we're still writing this doc.  
-Stay tuned, it will appear in a day or two.
+So you want to do routing with your Redux app. You can use it with the react-router library. Redux will be the source of truth for your data and react-router will be the source of truth for your URL. In most of the cases, **it is fine** to have them separate unless if you need to time travel and rewind actions that triggers the change URL.
+
+## Redux-router, react-router-redux, react-router
+Before starting, let's clarify the different routing libraries.
+
+### Redux-router
+Redux-router is an **experimental** library, it lets you keep entirely the state of your url inside your redux store. It has the same API with React Router API but has a smaller community support than react-router.
+
+### React-router-redux
+react-router-redux creates binding between your redux app and react-router and it keeps them in sync. Without this binding, you will not be able to rewind the actions with Time Travel. Unless you need this, React-router and Redux can operates completely apart.
+
+
+## Installing react-router
+react-router is available on npm :
+
+`npm install --save react-router`
+
+## Configuring the fallback url
+
+Before implementing react-router, we need to configure our back-end. Indeed, our back-end is not currently aware of the declared routes in react-router. If you refresh your page or try to access directly an URL declared in react-router without having configured your back-end, you will get an 404. You will be first requesting a url that the back-end is not aware of, instead of asking to react-router. You need to configure a fallback URL, to serve index.html on an unknown URL so that in the front-end, react-router can handle the request.
+
+### Configuring express.js
+If you are serving your index.html from express.js :
+``` js
+  app.get('/*', (req,res) => {
+    res.sendfile(path.join(__dirname, 'index.html'))
+  })
+```
+
+### Configuring webpack-dev-server
+If you are serving your index.html from webpack-dev-server:
+You can add to your webpack.config.dev.js :
+```
+  devServer: {
+    historyApiFallback: true,
+  }
+```
+
+## Connecting the router with Redux App
+
+Along this chapter, we will be using the [Todos](https://github.com/reactjs/redux/tree/master/examples/todos) example. We recommend you to clone it while reading this chapter.
+
+The <Router /> component has to be a children of `<Provider/>` so the `<Router />` has access to the global `store`. `<Provider/>` is the higher-order component provided by react-redux that lets you bind Redux to React (see [Usage with React](../basics/UsageWithReact.md)).
```

Comment:
This is not quite correct. We put `<Router />` into `<Provider>` so that *route handlers* (including our `App`) will get access to the `store`. Router component itself doesn’t care.